docs/Coding/mapgen.rst

```diff
@@ -0,0 +1,569 @@
+  creating the river. If a river comes close to one of these tiles, it is
+  falling back to itself and is aborted.
+
+Forming a 4-river-grid: fatal
+  A river may not form a grid with four rivers tiles next to each other.
+
+Highlands:
+  Rivers must not flow up in mountains or hills if there are alternatives.
+
+Adjacent ocean:
+  Rivers must flow down to coastal areas when possible.
+
+Adjacent river:
+  Rivers must flow down to areas near other rivers when possible:
+
+Adjacent highlands:
+  Rivers must not flow towards highlands (terrains with a non-zero
+  ``mountainous`` property) if there are alternatives.
+
+Swamps:
+  Rivers must flow down in swamps when possible.
+
+Adjacent swamps:
+  Rivers must flow towards swamps when possible.
+
+Height map:
+  Rivers must flow in the direction which takes it to the tile with the lowest
+  value on the height map.
+
+If these rules haven't decided the direction, the random number generator gets
+the decision.
+
+Once a river has been formed, it is added to the map by adding the river extra
+as needed. If the river goes through terrains where rivers are forbidden, the
+terrain is simply changed. The whole process is repeated until enough tiles are
+covered with rivers according to the :ref:`river fraction <mapgen-fractions>`.
+
+.. _mapgen-island:
+
+Island-Based Generators
+-----------------------
+
+The ``ISLAND`` generator setting corresponds to three different code paths
+depending on the ``startpos`` setting. They all try to fill the map with
+islands, but of different sizes and in different numbers to match the desired
+starting positions. The general idea is to generate island shapes randomly and
+paste them somewhere on the map. If this doesn't work, the process is repeated
+with a slightly smaller island until a fitting size and shape is found. Once
+islands are created, the map is finalized by filling them with terrain and
+creating rivers.
+
+The three available island-based generators are as follows, selected according
+to the value of ``startpos``:
+
+``VARIABLE``:
+  This generator tries to create three types of islands: 70% of the land is
+  given to big islands, 20% to medium, and 10% to small islands. In an ideal
+  case, each player starts on a big island and gets one medium and one small
+  island. However, due to the random nature of the algorithm it is not
+  guaranteed that the smaller islands will be evenly distributed.
+
+  If it is unable to create all the big islands, this generator starts again
+  with a smaller size until they all fit, increasing the size of medium islands
+  accordingly. Medium and small islands are optional and are only generated if
+  they can be placed on the map.
+
+``DEFAULT`` or ``SINGLE``:
+  This generator also tries to create one big island per player. Their size
+  follows a slightly complicated formula. The available landmass is first
+  divided by the number of players. The islands are get one third of the
```

Comment:
s/are/all. I would also break up this into discrete sentences vs the semi-colon. Makes it easier to read.